### PR TITLE
fix!: Allow protocol in SASLAUTHD_LDAP_SERVER & adjust SASLAUTHD_LDAP_ default values

### LINKED ISSUE #1983: [BUG] SASLAUTHD_LDAP_SSL=1 does not set ldaps://
# Bug Report

### What is affected by this bug?

Setting the environmental variable SASLAUTHD_LDAP_SSL to "1" does not set "ldaps://" in /etc/saslauthd.conf.  This is breaking saslauthd from working as the parameter "ldap_servers" is not set to a proper URI.

### When does this occur?

When setting the environmental variable SASLAUTHD_LDAP_SSL to "1" 

### How do we replicate the issue?

Set the environmental variable SASLAUTHD_LDAP_SSL to "1".

### Actual Behavior

The parameter "ldap_servers" in the file /etc/saslauthd.conf is being set like the below when setting the environmental variable SASLAUTHD_LDAP_SSL to "1". 

```
ldap_servers: 10.10.10.10
```

I think the problem seems to be with the logic in https://github.com/docker-mailserver/docker-mailserver/blob/master/target/scripts/startup/setup-stack.sh:

```
if [[ -z ${SASLAUTHD_LDAP_SSL} ]] || [[ ${SASLAUTHD_LDAP_SSL} -eq 0 ]]
then
    SASLAUTHD_LDAP_PROTO='ldap://' || SASLAUTHD_LDAP_PROTO='ldaps://'
fi
```

When the environmental variable SASLAUTHD_LDAP_SSL to "1", the logic above will set SASLAUTHD_LDAP_PROTO to "" (i.e empty).  Later in the code when SASLAUTHD_LDAP_PROTO is used, this parameter will be "" (i.e empty) and so SASLAUTHD_LDAP_SERVER will only be printed.

```
if [[ ! -f /etc/saslauthd.conf ]]  then
    _notify 'inf' 'Creating /etc/saslauthd.conf'
    cat &gt; /etc/saslauthd.conf &lt;&lt; EOF
ldap_servers: ${SASLAUTHD_LDAP_PROTO}${SASLAUTHD_LDAP_SERVER}
```

If I am not mistaken, I think this logic needs to be updated so that it properly sets SASLAUTHD_LDAP_PROTO='ldaps://' when the environmental variable SASLAUTHD_LDAP_SSL to "1". 

### Expected Behavior

The parameter "ldap_servers" in the file /etc/saslauthd.conf should be set like the below when setting the environmental variable SASLAUTHD_LDAP_SSL to "1".

ldap_servers: ldaps://10.10.10.10

## Your Environment

- version: `v`
- available RAM: `GB`
- Docker version: `v`
- docker-compose version: `v`

### Environment Variables

```CFG
SASLAUTHD_LDAP_SSL=1 
```

### Relevant Stack Traces

```BASH
# BEGIN

# END
```

### PR DESCRIPTION
**Pull request by moqmar**
# Description

Deprecate SASLAUTHD_LDAP_SSL by allowing the protocol in SASLAUTHD_LDAP_SERVER & use more sane defaults for some SASLAUTHD_LDAP variables.

Fixes #1983

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project (I used the style of the code around my changes, I think there are some inconsistencies with the default values in the saslauthd section compared to the other code)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] ~~If necessary I have added tests that prove my fix is effective or that my feature works~~
- [ ] New and existing unit tests pass locally with my changes (will have to test this tomorrow when I have a better internet connection)

### CODE REVIEW

**Comment by aendeavor** [comment]
This PR is to be merged before the release of v10 as it contains breaking changes:) Just FYI

**Comment by aendeavor** [comment]
@moqmar please reach back to us when this PR is ready to be reviewed, so we can take action:)

**Comment by moqmar** [comment]
Alright, I'd say if the checks run fine now then this is done!

Note that there are currently no regression tests for this as there's only one setup_file, so that would require big changes to the testing methodology.

File: test/mail_with_ldap.bats
Comment (polarathene) [inline test/mail_with_ldap.bats:44]: If this is being removed elsewhere, why is it added here now? The associated commit message wasn't very clear.

Presumably for testing the case in `setup-stack.sh`?

File: target/scripts/startup/setup-stack.sh
Comment (wernerfred) [inline target/scripts/startup/setup-stack.sh:633]: Why is `SASLAUTH_LDAP_SSL` still used here? Thought you tried to eliminate it?

Comment (moqmar) [reply target/scripts/startup/setup-stack.sh:633]: It's still there for better backwards compatibility - shall I add a warning there or shall I remove it completely?

File: test/mail_with_ldap.bats
Comment (moqmar) [reply test/mail_with_ldap.bats:44]: `SASLAUTHD_LDAP_SSL=0` together with the protocol being specified would previously break, with this I basically check if it's ignored in favor of the protocol in `SASLAUTHD_LDAP_SERVER`.

File: target/scripts/startup/setup-stack.sh
Comment (wernerfred) [reply target/scripts/startup/setup-stack.sh:633]: Imho as we are about to release v10 - a major release - we warn about breaking changes and that people have to adapt their setup accordingly. As this is such a change i would vote for removing it completely as otherwise we will have the problem at a later point in time.

cc @aendeavor

Comment (aendeavor) [reply target/scripts/startup/setup-stack.sh:633]: > As this is such a change i would vote for removing it completely as otherwise we will have the problem at a later point in time.

I agree. We can safely remove it.

File: test/mail_with_ldap.bats
Comment (polarathene) [reply test/mail_with_ldap.bats:44]: If that's being kept there should be a comment about it for context when another contributor runs across it.

Seems like it's being asked to drop from `setup-stack.sh` though, so perhaps has no relevance here anymore either?

Comment (aendeavor) [reply test/mail_with_ldap.bats:44]: This should be removed as well, see my comment below :)

**Review by wernerfred — APPROVED** [review]
LGTM 👍🏻 

Anything else needed from your side @moqmar?

If not @docker-mailserver/maintainers the next one to approve pls merge this so we can start feature freeze and testing for `v10.0.0`

**Comment by aendeavor** [comment]
> LGTM 👍🏻
> 
> Anything else needed from your side @moqmar?
> 
> If not @docker-mailserver/maintainers the next one to approve pls merge this so we can start feature freeze and testing for `v10.0.0`

I have opened #1996 which does not change anything else than the layer count. After this PR is merged, we can begin with the feature freeze. We should merge #1996 before releasing v10 since its a safe improvement, see the PR description.

**Review by aendeavor — APPROVED** [review]

**Comment by wernerfred** [comment]
We currently have a failing CI on master. The merged PR check passed though. I retriggered the check, maybe wait with merge until it passes

**Comment by aendeavor** [comment]
> We currently have a failing CI on master. The merged PR check passed though. I retriggered the check, maybe wait with merge until it passes

Just in time, I was about to merge this

**Comment by wernerfred** [comment]
Tests passed. Now build and publish is in Progress. You can merge imho

**Comment by aendeavor** [comment]
> Tests passed. Now build and publish is in Progress. You can merge imho.
> Seems Like the First Run had a Timing/CI Environment issue

I will merge this when CI has fully passed :D

**Pull request merged**
